core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -338,6 +363,7 @@ DEFINE_UNARY_FUNCTION_EVAL(log2, 2);
 DEFINE_UNARY_FUNCTION_EVAL(log1p, 2);
 
 DEFINE_UNARY_FUNCTION_EVAL(sqrt, 2);
+DEFINE_UNARY_FUNCTION_EVAL_CUSTOM(rsqrt, 2, 1 / std::sqrt(x));
```

Comment:
The type of `1` here is not clean. `decltype(x)(1)` will fix it for `float` and `double` but not for integer types. Nevertheless, it will work because `std::sqrt` returns `double`. Any suggestions?

Reply:
`decltype(std::sqrt(x))(1) / std::sqrt(x)` should do the trick.